zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+
+    issuance_txid_digest
+    ├── issueActions
+    └── issuerVerificationKey
+
+In the specification below, nodes of the tree are presented in depth-first order.
+
+issuance_txid_digest
+--------------------
+A BLAKE2b-256 hash of the following values ::
+
+   T.1: issueActions            (field encoding bytes)
+   T.2: issuerVerificationKey   (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdOrcZSAIssue"
+
+T.1 issueActions
+````````````````
+A byte encoding of Issue Action information for all Issue Actions belonging to the transaction. For each Action, the following elements are included in the hash::
+
+   T.1a: notes                   (field encoding bytes)
+   T.1b: assetDescription        (field encoding bytes)
+   T.1c: isFinalized             (1 byte)
+
+
+T.1a: notes
+'''''''''''
+A byte encoding of Note information for all Notes belonging to the Action. For each Note, the following elements are included in the hash::
+
+   T.1a.1: recipient                    (field encoding bytes)
+   T.1a.2: value                        (field encoding bytes)
+   T.1a.3: asset                        (field encoding bytes)
+   T.1a.4: rho                          (field encoding bytes)
+   T.1a.5: rseed                        (field encoding bytes)
+
+
+T.1a.1: recipient
+.................
+Raw Address encoded as specified in [Zcash Protocol Spec § 5.6.4.2: Orchard Raw Payment Addresses].
+
+T.1a.2: value
+.............
+Note value encoded as little-endian 8-byte representation of u64 raw value.
+
+T.1a.3: asset
+.............
+Asset ID encoded as 32-byte representation of Pallas point.
+
+T.1a.4: rho
+...........
+Nullifier encoded as 32-byte representation of Pallas point.
+
+T.1a.5: rseed
+.............
+The ZIP 212 32-byte seed randomness for a note.
+
+T.1b: assetDescription
+''''''''''''''''''''''
+UTF-8 encoding of the asset description string.
+
+T.1c: isFinalized
+'''''''''''''''''
+1-byte representation of a boolean flag that is set to 'true' if the asset type was finalized in this action and 'false' otherwise. 'True' is represented as 1, 'false' as 0.
+
+
+T.2 issuerVerificationKey
+`````````````````````````
+A byte encoding of issuer verification key for the bundle as defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components].
+
+
+Security and Privacy Considerations
+===================================
+
+**Issuer Key or AssetId Compromise**
+
+The design of this protocol does not allow for a rotation of the `issuerValidatingKey`, that would allow for replacing the key of a specific asset (see Future Work). In case of compromise, the following actions are recommended:
+- If an asset ID is compromised (and not the issuer verification key), the `finalize` boolean for that asset should be set to `0` and a new `AssetId` generated instead.
```

Comment:
Does this mean a bridge for a particular assetId is compromised? That is not described or specified in this document. This document would indicate `assetId` compromise involves some sort of preimage or collision attack involving GroupHash

Reply:
We removed the portions that were not relevant to this specification.